### ISSUE TITLE
Plugin errors in Android Studio 4.2 beta1

### ISSUE THREAD

**Issue by skolson** [mannodermaus/android-junit5#230]
Using de.mannodermaus.gradle.plugins:android-junit5:1.6.2.0 in Android Studio 4.2 beta1 in a multi-module project with:

-  Gradle 6.7.1
-  kotlin 1.4.20
- JUnit 5.7.0 
- com.android.tools.build:gradle:4.2.0-beta01
- androidx.test:runner:1.3.0 
- de.mannodermaus.junit5:android-test-core:1.2.0
- de.mannodermaus.junit5:android-test-runner:1.2.0

I'm seeing two issues:
1) Running an individual instrumented test works, but @BeforeAll function not invoked 
2) in the android block of build.gradle.kts, attempting this testOptions block:
```
    testOptions {
        junitPlatform {
            instrumentationTests {
                // Whether or not to validate configuration of instrumentation tests in this build
                integrityCheckEnabled = true
            }
        }
    }
```
The junitPlatform block shows as a syntax error even though the de.mannodermaus.android-junit5 is present and working in the plugins block.

Without the testOptions block, and manually invoking the @BeforeAll function, the instrumented test runs fine, and debugs fine using JUnit5 as far as I can tell.
I know AS 4.2 beta 1 is brand new (and a beta), as is the android gradle plugin, so issues are not surprising.  If there is anything additional I can do to help diagnose this, let me know.

**Comment by skolson** [comment]
Just tried 1.7.0.0-SNAPSHOT from 11/5/2020, it has the same two issues.

**Comment by mannodermaus** [comment]
Thanks for testing against the snapshot as well. I take it you're using Gradle 7.1? I wasn't even aware that exists already. Not sure if the Android Gradle Plugin itself is ready for that yet, but maybe the Kotlin DSL has some quirks with Gradle 7. You could try `isIntegrityCheckEnabled = true` as an alternative since it might try and rewrite the boolean flag with the prefix

**Comment by skolson** [comment]
LOL I didn't notice I typo'd gradle - thats supposed to be 6.7.1!

**Comment by skolson** [comment]
I found the cause of the issue with @BeforeAll. The test class was missing this annotation:
`@TestInstance(TestInstance.Lifecycle.PER_CLASS)`
So the only remaining issue is why juniPlatform is unresolved in build.gradle.kts despite the plugin working.

**Comment by skolson** [comment]
Nothing I've tried allows the IDE to recognize junitPlatform within testOptions.  The plugin that adds that extension to the DSL is definitely applied and working, Looks like a defect to me :-)  So if you disagree, close this as I have no questions left...

**Comment by mannodermaus** [comment]
I've had a similar experience with a different plugin before, where the provided DSL was not recognized at all and I had to use a different means to configure it. Might be a Kotlin DSL/Android Studio issue, rather than something within android-junit5.

As a workaround, you can forego the DSL sugar and access the extension directly with Kotlin DSL's `the` method:

```kotlin

android {
  testOptions {
    with(the<AndroidJUnitPlatformExtension>()) {
      filters {
        includeEngines("my-engine")
      }
    }
  }
}
```

**Comment by mannodermaus** [comment]
Seems like the extensibility of types has been limited in the more recent versions of the Android Gradle Plugin. It doesn't seem possible anymore to add a Gradle extension on top of an arbitrary `ExtensionAware` type. Going forward, the `junitPlatform` DSL will be moved to the project instead, which should resolve this issue. I'll update this ticket as soon as a snapshot build is available for testing.

**Comment by mannodermaus** [comment]
This was resolved with the 1.7.1.1 release of the plugin, which moved the location of the extension to the `Project` rather than the `TestOptions` block of the Android plugin. After updating to that version, either of the following will work (though the first version is more canonical):

```kotlin
junitPlatform {
  // Configure
}
```

```kotlin
android {
  testOptions {
    junitPlatform {
      // Configure
    }
}
```
